boards/arm/gd32e103v_eval/doc/index.rst

```diff
@@ -0,0 +1,212 @@
+.. _gd32e103v_eval:
+
+GigaDevice GD32E103V-EVAL
+#########################
+
+
+Overview
+********
+
+The GD32E103V-EVAL board is a hardware platform that enables design and debug
+of the GigaDevice E103 Cortex-M4F High Performance MCU.
+
+The GD32E103VB features a single-core ARM Cortex-M4F MCU which can run up
+to 120-MHz with flash accesses zero wait states, 128kiB of Flash, 32kiB of
+SRAM and 80 GPIOs.
+
+.. image:: img/gd32e103v_eval.png
+     :width: 800px
+     :align: center
+     :alt: gd32e103v_eval
+
+
+Hardware
+********
+
+- USB interface with mini-USB connector
+- 4 user LEDs
+- 4 user push buttons
+- Reset Button
+- ADC connected to a potentiometer
+- 2 DAC channels
+- GD25Q16 2Mib SPI Flash
+- AT24C02C 2KiB EEPROM
+- 3.2 TFT LCD (320x240)
+- PCM1770 Stereo DAC with Headphone Amplifier
+- GD-Link interface
+
+  - CMSIS-DAP swd debug interface over USB HID.
+
+- 2 CAN port(support CAN-FD)
+
+  - This function is not available in this board due to hardware issues, please check ``GD32C103`` .
+
+For more information about the GD32E103 SoC and GD32E103V-EVAL board:
+
+- `GigaDevice Cortex-M4F High Performance SoC Website`_
+- `GD32E103 Datasheet`_
+- `GD32E103 Reference Manual`_
+- `GD32E103V Eval Schematics`_
+- `GD32 ISP Console`_
+
+
+Supported Features
+==================
+
+The board configuration supports the following hardware features:
+
++-----------+------------+-----------------------+
+| Interface | Controller | Driver/Component      |
++===========+============+=======================+
+| NVIC      | on-chip    | nested vectored       |
+|           |            | interrupt controller  |
++-----------+------------+-----------------------+
+| SYSTICK   | on-chip    | system clock          |
++-----------+------------+-----------------------+
+| UART      | on-chip    | serial port-polling   |
```

Comment:
suggestion: use a structure like this, so that table points to relevant compatibles/Kconfig options

https://github.com/zephyrproject-rtos/zephyr/blob/58234d490c32811860ffc818d662730448c713a7/boards/arm/gd32f450i_eval/doc/index.rst?plain=1#L56-L70